src/user/user.service.ts

```diff
@@ -24,8 +24,22 @@ export class UserService {
   ) {}
 
   async createUser(dto: CreateUserDto) {
-    const role = await this.roleService.getRoleByValue(ROLE[ROLE.player]);
+    let role;
+    if(dto.registered) {
+      role = await this.roleService.getRoleByValue(ROLE[ROLE.player]);
```

Comment:
why are you still using that query, given migrations are run, you can safely assume you definitely have ones in the db, so just use roleId